source/simulator/parameters.cc

```diff
@@ -858,6 +888,24 @@ namespace aspect
                    ExcMessage("Invalid input parameter file: Too many entries in List of normalized fields"));
     }
     prm.leave_subsection ();
+
+
+    // now also get the parameters related to material model averaging
+    prm.enter_subsection ("Material model");
+    {
+      material_averaging_for_linear_systems
+        = MaterialModel::MaterialAveraging::parse_averaging_operation_name
+          (prm.get ("Material averaging for linear systems"));
+      material_averaging_for_preconditioners
+        = MaterialModel::MaterialAveraging::parse_averaging_operation_name
+          (prm.get ("Material averaging for preconditioners"));
+
+      // implement the special rule that if the linear systems use some
+      // averaging, then the preconditioner needs to use the same
+      if (material_averaging_for_linear_systems != MaterialModel::MaterialAveraging::none)
```

Comment:
Turns out that this doesn't work -- you can't precondition a non-averaged matrix with an averaged preconditioner. It looks like I overthought the issue. I will just abandon this issue.